NetworkPkg/UefiPxeBcDxe/GoogleTest/UefiPxeBcDxeGoogleTest.inf

```diff
@@ -23,7 +23,6 @@ VERSION_STRING = 1.0
   PxeBcDhcp6GoogleTest.h
   ../PxeBcDhcp6.c
   ../PxeBcSupport.c
-  ../../../MdePkg/Test/Mock/Library/GoogleTest/Protocol/MockRng.cpp

```

Comment:
This is one of the reasons I wanted to move away from having cpp files for protocol/PPI mocks. But are you saying we can't include `../PxeBcDhcp6.c` this way?

Reply:
Correct.  Use if .. in INF is not allowed

Reply:
@mdkinney  - your statement doesn't match the agreed upon practice or usage of "Host-Based Unit Tests for a Library/Driver (PEI/DXE/SMM) implementation" tests.  See the layout and the table we all agreed upon here: https://github.com/tianocore/edk2/tree/master/UnitTestFrameworkPkg#example-directory-tree  

If you look at `SpecificLibDxe` you see that we put the test code in a nested folder with test specific code and inf.  This allows the inclusion of the real code under test by using ".." syntax.  I agree that using relative pathing to go outside of your module folder is unacceptable but for the "../PxeBcDhcp6.c" line listed above this is agreed upon practice and in fact the best way to do it.   The MockRng.cpp is invalid and the point of this PR was to remove that pattern.  

I was looking through the edk2 codebase for examples and noticed that we do not have specific implementation tests.  There are a couple that but I think they are incorrectly implemented and/or not really specific implementation tests.  See  https://github.com/tianocore/edk2/blob/master/MdeModulePkg/Library/DxeResetSystemLib/UnitTest/DxeResetSystemLibUnitTestHost.inf but this is actually wrong because it is testing the library class interface rather than a specific instance and the way it is setup is fragile and depends on build logic being configured correctly rather than self-contained within the INF.  If the test really wanted to test the instance and have access to specific variables or private functions (the point of an instance test) it would need access to them which is not available as external lib.  Happy to discuss more if this isn't clear.

Reply:
@mdkinney If there are no further comments, resolving this thread.

Reply:
I can't recall.  Did we discuss moving the unit test INF file up one directory so no .. are required for files in the same module?